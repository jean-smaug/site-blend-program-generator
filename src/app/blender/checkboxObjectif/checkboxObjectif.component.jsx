import React, { Component } from 'react';
import PropType from 'prop-types';
import { connect } from 'react-redux';
import _ from 'lodash';
import * as formActions from '../blender.action';

/**
 * Component for one objectif's checkbox
 */

export class CheckboxObjectifComponent extends Component {
  constructor(props) {
    super(props);
    this.toggleCheckbox = this.toggleCheckbox.bind(this);
  }

  toggleCheckbox(e) {
    if (e.target.checked) this.props.addObjectif(this.props.item.id);
    else this.props.removeObjectif(this.props.item.id);
  }

  render() {
    return (
      <div>
        <div>
          <input
            onChange={this.toggleCheckbox}
            name={this.props.item.id}
            type="checkbox"
            checked={_.includes(this.props.objectifs, this.props.item.id)}
          />
          {this.props.item.libelle}
        </div>
      </div>
    );
  }
}

CheckboxObjectifComponent.propTypes = {
  addObjectif: PropType.func.isRequired,
  removeObjectif: PropType.func.isRequired,
  item: PropType.shape({
    id: PropType.number.isRequired,
    libelle: PropType.string.isRequired,
  }).isRequired,
  objectifs: PropType.arrayOf({
    id: PropType.number.isRequired,
    libelle: PropType.string.isRequired,
  }).isRequired,
};

const mapStateToProps = state => ({
  objectifs: state.form.objectifs,
});

const mapDispatchToProps = dispatch => ({
  addObjectif: (objectif) => {
    dispatch(formActions.addObjectif(objectif));
  },
  removeObjectif: (objectif) => {
    dispatch(formActions.removeObjectif(objectif));
  },
});

const CheckboxObjectif = connect(mapStateToProps, mapDispatchToProps)(
  CheckboxObjectifComponent,
);

export default CheckboxObjectif;
