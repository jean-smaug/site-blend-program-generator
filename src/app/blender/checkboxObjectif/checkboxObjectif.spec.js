import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import { CheckboxObjectifComponent } from './checkboxObjectif.component';

const item = {
  id: '1',
  libelle: 'tech',
};

describe('checkboxDomain.component', () => {
  let wrapper;
  let store;
  const mockStore = configureStore();
  const initialState = {};

  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = shallow(<CheckboxObjectifComponent item={item} store={store} />);
  });

  it('should render component', () => {
    expect(wrapper.getNodes()).toMatchSnapshot();
  });
});
