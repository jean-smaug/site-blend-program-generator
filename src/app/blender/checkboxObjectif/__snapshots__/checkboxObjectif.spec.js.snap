// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkboxDomain.component === SNAPSHOT === 1`] = `
<div>
  <div>
    <input
      checked={false}
      name="1"
      onChange={[Function]}
      type="checkbox"
    />
    tech
  </div>
</div>
`;
